### DISCUSSION TITLE
Exif Orientation

### DISCUSSION THREAD

**Discussion by AchimGaedkeLynker** [kornia/kornia#3025] [Questions]
Hi there!

I use annotations from [CVAT](https://docs.cvat.ai/about/) to train Object Detection models. The annotations (bounding boxes) returned by the CVAT SDK are in the XYXY format (unit: pixels).

For a while I got some bad training results, and then I realized that CVAT shows the images after rotating them according to the [Exif Orientation](https://sirv.com/help/articles/rotate-photos-to-be-upright/) flag, so the image loaded by the PIL/Kornia/pytorch ML software stack does not fit the annotation for orientation values > 1.

I fixed the problem using `PIL.Image.Image.getexif()` - or `PIL.ImageOps.exif_transpose()`

For example:

```python
def load_image(im_path):
    im_data = kornia.io.load_image(im_path, desired_type=kornia.io.ImageLoadType.RGB32, device="cuda")
    exif_orientation = PIL.Image.open(im_path).getexif().get(ExifTags.Base.Orientation, 1)
    return im_data.rot90(k={1: 0, 8: 1, 3: 2, 6: 3}[exif_orientation], dims=[1,2])
```

But that opens the file twice (I believe it reads the file completely only once) and uses both Kornia and PIL.

Can I achieve the same (read out exif metadata or get the rotated/flipped image according to the Exif Orientation) with functions only from the Kornia / Kornia-Rust packages?

Many thanks!

**Comment by edgarriba** [comment]
> Can I achieve the same (read out exif metadata or get the rotated/flipped image according to the Exif Orientation) with functions only from the Kornia / Kornia-Rust packages?

@AchimGaedkeLynker not that I'm aware of right now. Is it possible that to share an image sample that contains a valid exif ? I can try to something in kornia-rs. Probably something useful that can be done is have a function that given the images in raw bytes we can get the exif  data and then you can use that piece of info to transpose directly the tensor in the gpu.

**Reply by edgarriba** [reply]
also probably the best place to put this is issue is in the rust repo as kornia pytorch wraps in the end that
https://github.com/kornia/kornia-rs/issues

**Comment by AchimGaedkeLynker** [comment]
Took my phone, holding it upright and sidewise... results here:

Upright
![Logo](https://github.com/user-attachments/assets/d2aec8c0-8a9e-4862-9d94-7ccb12094689)

Sidewise
![Logo90CW](https://github.com/user-attachments/assets/cb4c7d15-8167-4a2c-9ad9-63a6d82ecb4d)

Use https://jimpl.com/ to check the Full Metadata - yes a lot of image viewing software reads and applies the EXIF orientation - that makes the problem less obvious
